src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmRpoServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.darts.arm.service.impl;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.arm.service.ArmRpoService;
+import uk.gov.hmcts.darts.authorisation.component.UserIdentity;
+import uk.gov.hmcts.darts.common.entity.ArmRpoExecutionDetailEntity;
+import uk.gov.hmcts.darts.common.entity.ArmRpoStatusEntity;
+import uk.gov.hmcts.darts.common.exception.DartsException;
+import uk.gov.hmcts.darts.common.helper.CurrentTimeHelper;
+import uk.gov.hmcts.darts.common.repository.ArmRpoExecutionDetailRepository;
+
+@Service
+@AllArgsConstructor
+public class ArmRpoServiceImpl implements ArmRpoService {
```

Comment:
There are unit tests for every method but I have split them so they have their individual unit tests per method (e.g. ArmRpoApiAddAsyncSearchTest) so that the tests for each of these implements can be worked on in isolation so you don't end up with 13 success test and at least another 13 failure paths in one test

Reply:
I believe that was for  ArmRpoApiImpl not for ArmRpoServiceImpl

Reply:
Added tests for ArmRpoServiceImpl